astropy/coordinates/tests/test_representation.py

```diff
@@ -1073,3 +1086,208 @@ def test_combine_xyz():
+        # pass through the differentials
+        for name in REPRESENTATION_CLASSES:
+            new_rep = rep1.represent_as(REPRESENTATION_CLASSES[name])
+            assert new_rep.get_name() == REPRESENTATION_CLASSES[name].get_name()
+            assert len(new_rep.differentials) == 1
+            assert new_rep.differentials[0] == diff
+
+        with pytest.raises(ValueError) as excinfo:
+            rep1.represent_as('name')
+        assert 'use frame object' in str(excinfo.value)
+
+    def test_getitem(self):
+
+        d = CartesianDifferential(d_x=np.arange(10) * u.m/u.s,
+                                  d_y=-np.arange(10) * u.m/u.s,
+                                  d_z=1. * u.m/u.s)
+        s = CartesianRepresentation(x=np.arange(10) * u.m,
+                                    y=-np.arange(10) * u.m,
+                                    z=3 * u.km,
+                                    differentials=d)
+
+        s_slc = s[2:8:2]
+        s_dif = s_slc.differentials[0]
+
+        assert_allclose_quantity(s_slc.x, [2, 4, 6] * u.m)
+        assert_allclose_quantity(s_slc.y, [-2, -4, -6] * u.m)
+        assert_allclose_quantity(s_slc.z, [3, 3, 3] * u.km)
+
+        assert_allclose_quantity(s_dif.d_x, [2, 4, 6] * u.m/u.s)
+        assert_allclose_quantity(s_dif.d_y, [-2, -4, -6] * u.m/u.s)
+        assert_allclose_quantity(s_dif.d_z, [1, 1, 1] * u.m/u.s)
+
+    def test_transform(self):
+        d = CartesianDifferential(d_x=[1, 2] * u.km/u.s,
+                                  d_y=[3, 4] * u.km/u.s,
+                                  d_z=[5, 6] * u.km/u.s)
+        s1 = CartesianRepresentation(x=[1,2] * u.kpc,
+                                     y=[3,4] * u.kpc,
+                                     z=[5,6] * u.kpc,
+                                     differentials=d)
+
+        matrix = np.array([[1,2,3], [4,5,6], [7,8,9]])
+
+        with pytest.raises(TypeError):
+            s1.transform(matrix)
+
+    def test_with_differentials(self):
+        # make sure with_differential correctly creates a new copy with the same
+        # differential
+        cr = CartesianRepresentation([1, 2, 3]*u.kpc)
+        diff = CartesianDifferential([.1, .2, .3]*u.km/u.s)
+        cr2 = cr.with_differentials((diff,))
+        assert cr.differentials != cr2.differentials
+        assert cr2.differentials[0] is diff
+
+        # make sure its a copy and not a view
+        assert cr.x == cr2.x
+        cr2.x.ravel()[:] = 10*u.kpc
+        assert cr.x != cr2.x
+
+        # make sure it works even if a differential is present already
+        diff2 = CartesianDifferential([.1, .2, .3]*u.m/u.s)
+        cr3 = CartesianRepresentation([1, 2, 3]*u.kpc, differentials=diff)
+        cr4 = cr3.with_differentials((diff2,))
+        assert cr4.differentials[0] != cr3.differentials[0]
+        assert cr4.differentials[0] == diff2
+
+        # also ensure a *scalar* differential will works
```

Comment:
I implemented the `_validate_differentials` method so that if a scalar is passed in, it calls `._get_deriv_key()` to get the appropriate dict key and then loads it into the dictionary. Might be easier than always having to pass in a dictionary?